website/pages/en/cookbook/timeseries.mdx

```diff
@@ -0,0 +1,179 @@
+  id: Int8!
+  timestamp: Timestamp!
+  sum: BigDecimal! @aggregate(fn: "sum", arg: "price")
+}
+```
+
+In this example, Stats aggregates the price field from Data over hourly and daily intervals, computing the sum.
+
+### Querying Aggregated Data
+
+Aggregations are exposed via query fields that allow filtering and retrieval based on dimensions and time intervals.
+
+Example:
+
+```graphql
+{
+  tokenStats(
+    interval: "hour"
+    where: { token: "0x1234567890abcdef", timestamp_gte: "1704164640000000", timestamp_lt: "1704251040000000" }
+  ) {
+    id
+    timestamp
+    token {
+      id
+    }
+    totalVolume
+    priceUSD
+    count
+  }
+}
+```
+
+### Using Dimensions in Aggregations
+
+Dimensions are non-aggregated fields used to group data points. They enable aggregations based on specific criteria, such as a token in a financial application.
+
+Example:
+
+### Timeseries Entity
+
+```graphql
+type TokenData @entity(timeseries: true) {
+  id: Int8!
+  timestamp: Timestamp!
+  token: Token!
+  amount: BigDecimal!
+  priceUSD: BigDecimal!
+}
+```
+
+### Aggregation Entity with Dimension
+
+```graphql
+type TokenStats @aggregation(intervals: ["hour", "day"], source: "TokenData") {
+  id: Int8!
+  timestamp: Timestamp!
+  token: Token!
+  totalVolume: BigDecimal! @aggregate(fn: "sum", arg: "amount")
+  priceUSD: BigDecimal! @aggregate(fn: "last", arg: "priceUSD")
+  count: Int8! @aggregate(fn: "count", cumulative: true)
+}
+```
+
+- Dimension Field: token groups the data, so aggregates are computed per token.
+- Aggregates:
+  - totalVolume: Sum of amount.
+  - priceUSD: Last recorded priceUSD.
+  - count: Cumulative count of records.
+
+### Aggregation Functions and Expressions
+
+Supported aggregation functions:
+
+- sum
+- count
+- min
+- max
+- first
+- last
+
+The arg in @aggregate can be:
+
+- A field name from the timeseries entity.
+- An expression using fields and constants.
+
+Examples of Aggregation Expressions:
+
+- Sum Token Value: @aggregate(fn: "sum", arg: "priceUSD \_ amount")
+- Maximum Positive Amount: @aggregate(fn: "max", arg: "greatest(amount0, amount1, 0)")
+- Conditional Sum: @aggregate(fn: "sum", arg: "case when amount0 > amount1 then amount0 else 0 end")
+
+Supported operators and functions include basic arithmetic (+, -, \_, /), comparison operators, logical operators (and, or, not), and SQL functions like greatest, least, coalesce, etc.
+
+### Query Parameters
+
+- interval: Specifies the time interval (e.g., "hour").
+- where: Filters based on dimensions and timestamp ranges.
+- timestamp_gte / timestamp_lt: Filters for start and end times (microseconds since epoch).
+
+### Notes
+
+- Sorting: Results are automatically sorted by timestamp and id in descending order.
+- Current Data: An optional current argument can include the current, partially filled interval.
+
+### Important Considerations
+
+- Immutable Data: Timeseries data cannot be altered once written, ensuring data integrity and simplifying indexing.
```

Comment:
It might be useful to place this closer to the top bc it strikes me as a warning

Reply:
+1